src/main/java/io/confluent/connect/jdbc/sink/BufferedRecords.java

```diff
@@ -219,6 +220,9 @@ public BufferedRecords(
   private Optional<Long> executeUpdates() throws SQLException {
     Optional<Long> count = Optional.empty();
     for (int updateCount : updatePreparedStatement.executeBatch()) {
+      if (updateCount == Statement.EXECUTE_FAILED) {
```

Comment:
I assume that if this error code is returned that there is no further context (like as an exception) to be retrieved from the driver. I don't see any relevant details to include in the error message nearby, except for the number of records or records themselves, so this might already be the best error message possible.

Reply:
That's right. I will explore a bit more to see if there is anything that the driver could give back. The driver might just throw an exception in many cases or return this error code. We were not handling this scenario at all and we were  relied on the count reconcile to throw an exception. If we don't do a reconcile, we need this explicit check.

Reply:
```
If one of the commands in a batch update fails to execute properly, this method throws a BatchUpdateException, and a JDBC driver may or may not continue to process the remaining commands in the batch. However, the driver's behavior must be consistent with a particular DBMS, either always continuing to process commands or never continuing to process commands. If the driver continues processing after a failure, the array returned by the method BatchUpdateException.getUpdateCounts will contain as many elements as there are commands in the batch, and at least one of the elements will be the following:
A value of EXECUTE_FAILED -- indicates that the command failed to execute successfully and occurs only if a driver continues to process commands after a command fails
```
Trying to understand the wording here. If one of the command fails, then the `executeBatch` method throws a `BatchUpdateException` and `BatchUpdateException.getUpdateCounts` returns an array of INT which contains `EXECUTE_FAILED` for the commands in batch that failed. 

So from my understanding this statement will never be true.

Reply:
maybe the right thing to do would be to get the array from `BatchUpdateException` if it fails or rely on the response from the actual method for success cases and for drivers that don't do this per the spec.

Reply:
By not doing anything here, I think `JdbcSinkTask` attempts to retry a few times and fails the task after retry expiry, logging the update counts and `EXECUTE_FAILED` values.

Reply:
I am wrong. I don't see the updateCounts explicitly included in the Exception message. It would be better to log it, rethrow the same exception and not throw a `ConnectException` to retain the existing retry behavior.